sdk/core/Azure.Core.TestFramework/src/Instrumentation/DiagnosticScopeValidatingInterceptor.cs

```diff
@@ -202,6 +203,12 @@ internal static async ValueTask<T> ValidateDiagnosticScope<T>(Func<ValueTask<(T
                             throw new InvalidOperationException($"All diagnostic scopes should have 'az.namespace' attribute, make sure the assembly containing **ClientOptions type is marked with the AzureResourceProviderNamespace attribute specifying the appropriate provider. This attribute should be included in AssemblyInfo, and can be included by pulling in AzureResourceProviderNamespaceAttribute.cs using the AzureCoreSharedSources alias.");
                         }
 
+                        Activity parent = e.Activity.Parent;
+                        if (parent != null && !parent.Tags.Any(tag => tag.Key == "kind" && (tag.Value == "client" || tag.Value == "internal")))
+                        {
+                            throw new InvalidOperationException($"Client diagnostic scopes should not be nested except for backward compatibility reasons. Use ClientDiagnostics constructor overload to suppress nested scopes.");
```

Comment:
So if you do need to set the flag to False, how do you avoid this error? Wouldn't we need to check the value of the property to determine whether this check should be done?

Reply:
Does this make the check here no longer necessary - https://github.com/Azure/azure-sdk-for-net/blob/d82b2ce6271cb4017f4037496fb973a9a3c78490/sdk/core/Azure.Core.TestFramework/src/ClientDiagnosticListener.cs#L147-L153

Reply:
I reverted the change to this file for now.  The new check has to be conditional and depend on `DiagnsoticsOptions.AreNestedClientSpansEnabled` flag value.

But  I can't do it until the new Azure.Core is shipped (TestFramework uses package reference). There are some storage tests that fail with this check on.

And you're right. With this check, `A scope has already started for event '{producedDiagnosticScope.Name}` becomes unnecessary.